### ISSUE TITLE
Deleting primary shop email is possible and renders app unusable

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#3446]
### Expected behavior
You should not be able to remove the primary shop email and if it's not present it should be handled gracefully (somehow)

### Actual behavior
You can remove the primary contact email from a shop and when you do that login is no longer available and the server just throws errors and the app never loads

### Steps to reproduce the behavior
1. Login as admin
1. Go into Shops panels and remove the "primary contact".
1. Log out
1. Observe the app no longer loads

### Versions
```
Node: 8.9.1
NPM: 5.5.1
Meteor Node: 8.9.3
Meteor NPM: 5.5.1
Reaction CLI: 0.24.2
Reaction: 1.6.3
Reaction branch: master
Docker: 17.09.1-ce
```

**Comment by zenweasel** [comment]
Since you would have to go out of your way to remove the email this is low impact but once this happens there doesn't appear to be any way to recover through the UI. So we need to do both better validation on the shop pabel and also guard against the missing value.

**Comment by trojanh** [comment]
@aldeed will updating the emails array to following in Shop simpleSchema be the right solution for this?
[shop/simpleSchemas.js](https://github.com//reactioncommerce/reaction/blob/e39ff79c1c6bb5b0d19ef7993d5be053786abc41/src/core-services/shop/simpleSchemas.js#L677)
```
...
"emails": {
    type: Array,
    minCount: 1
  },
  "emails.$": {
    type: Email
  },
...
```

**Comment by aldeed** [comment]
@trojanh This is an old issue and I actually disagree that it is a problem. In 3.0 branch, I don't think it breaks anything other than emails do not send. We have discussed eventually having a general system for notifying admins about issues like this, things that ideally should be set up but most of the system will still work without it.

So I'm going to close this as "fixed in 3.0"/"not a bug"

**Comment by trojanh** [comment]
sure @aldeed  thanks, we now have one less issue to fix.
